test/index.js

```diff
@@ -21,334 +21,275 @@ const expect = Code.expect;
+    it('sets/gets following JS data types: Object, Array, Number, String, Date, RegExp', async () => {
 
         const client = new Catbox.Client(Mongo);
-        client.start((err) => {
+        await client.start();
 
-            expect(err).to.not.exist();
-            const key = { id: 'x', segment: 'test' };
-            const value = {
-                object: { a: 'b' },
-                array: [1, 2, 3],
-                number: 5.85,
-                string: 'hapi',
-                date: new Date('2014-03-07'),
-                regexp: /[a-zA-Z]+/
-            };
-
-            client.set(key, value, 500, (err) => {
+        const key = { id: 'x', segment: 'test' };
+        const value = {
+            object: { a: 'b' },
+            array: [1, 2, 3],
+            number: 5.85,
+            string: 'hapi',
+            date: new Date('2014-03-07'),
+            regexp: /[a-zA-Z]+/
+        };
 
-                expect(err).to.not.exist();
-                client.get(key, (err, result) => {
+        await client.set(key, value, 500);
 
-                    expect(err).to.equal(null);
-                    expect(result.item).to.deep.equal(value);
-                    done();
-                });
-            });
-        });
+        const result = await client.get(key);
+        expect(result.item).to.equal(value);
     });
 
-    it('fails setting an item circular references', (done) => {
+    it('fails setting an item circular references', async () => {
 
         const client = new Catbox.Client(Mongo);
-        client.start((err) => {
+        await client.start();
 
-            expect(err).to.not.exist();
-            const key = { id: 'x', segment: 'test' };
-            const value = { a: 1 };
-            value.b = value;
-            client.set(key, value, 10, (err) => {
+        const key = { id: 'x', segment: 'test' };
+        const value = { a: 1 };
+        value.b = value;
 
-                expect(err).to.exist();
-                done();
-            });
-        });
+        try {
+            await client.set(key, value, 10);
+        }
+        catch (err) {
+            expect(err).to.exist();
+        }
     });
 
-    it('ignored starting a connection twice on same event', (done) => {
+    it('ignored starting a connection twice on same event', async () => {
 
         const client = new Catbox.Client(Mongo);
         let x = 2;
-        const start = function () {
-
-            client.start((err) => {
+        const start = async function () {
 
-                expect(err).to.not.exist();
-                expect(client.isReady()).to.equal(true);
-                --x;
-                if (!x) {
-                    done();
-                }
-            });
+            await client.start();
+            expect(client.isReady()).to.equal(true);
+            --x;
+            if (!x) {
+            }
         };
 
-        start();
-        start();
+        await start();
```

Comment:
If you use await here - it won't call start twice at the same event as this test intend to.